src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -10,34 +10,69 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim collector = New Collector(_list)
             collector.Visit(Me)
-            Return _list.OfType(Of IOperation)()
+            Return _list
         End Function
 
         Private Class Collector
             Inherits BoundTreeWalkerWithStackGuard
 
-            Private nodes As List(Of BoundNode)
+            Private nodes As List(Of IOperation)
 
-            Public Sub New(nodes As List(Of BoundNode))
+            Public Sub New(nodes As List(Of IOperation))
                 Me.nodes = nodes
             End Sub
 
             Public Overrides Function Visit(node As BoundNode) As BoundNode
-                Me.nodes.Add(node)
+                Dim operation = TryCast(node, IOperation)
+                If operation IsNot Nothing Then
+                    Me.nodes.Add(operation)
+                    Select Case operation.Kind
+                        Case OperationKind.InvocationExpression
+                            Me.nodes.AddRange(DirectCast(operation, IInvocationExpression).ArgumentsInSourceOrder)
+                        Case OperationKind.ObjectCreationExpression
+                            Dim objCreationExp = DirectCast(operation, IObjectCreationExpression)
+                            Me.nodes.AddRange(objCreationExp.ConstructorArguments)
+                            Me.nodes.AddRange(objCreationExp.MemberInitializers)
+                        Case OperationKind.SwitchSection
+                            ' "Case Else" clause is not a bound node, so it needs to be added explicitly.
+                            Dim caseClauses = DirectCast(operation, ICase).Clauses
+                            If caseClauses.IsSingle Then
+                                Dim caseClause = caseClauses(0)
+                                If caseClause.CaseKind = CaseKind.Default Then
+                                    Me.nodes.Add(caseClause)
+                                End If
+                            End If
+                        Case OperationKind.ExpressionStatement
+                            Dim expression = DirectCast(operation, IExpressionStatement).Expression
+                            If expression.Kind = OperationKind.EventAssignmentExpression Then
+                                Me.nodes.Add(expression)
+                            End If
+                    End Select
+                End If
                 Return MyBase.Visit(node)
             End Function
 
+            ' Skip visiting `BoundAssignmentOperator` nodes (but Not their children) if they are used for initializing members in object creation.
+            ' The corresponding operations are covered by `IMemberInitializer`.
+            Public Overrides Function VisitObjectInitializerExpression(node As BoundObjectInitializerExpression) As BoundNode
+                For Each assignment As BoundAssignmentOperator In node.Initializers
+                    Me.Visit(assignment.Left)
+                    Me.Visit(assignment.LeftOnTheRightOpt)
```

Comment:
LeftOnTheRightOpt doesn't represent any expression from the original syntax and it has very special meaning. Is there a point in visiting it? BTW, should anything special be done when LeftOnTheRightOpt node is encountered while visiting assignment.Right?